packages/headless/src/features/debug/ranking-info-parser.ts

```diff
@@ -123,34 +241,34 @@ const parseTermsWeights = (
   if (!listOfTerms) {
     return null;
   }
-  const terms = {} as ListOfTermsWeights;
+  const terms: TermWeightReport = {};
   for (const term of listOfTerms) {
     const listOfWords = matchExec(term[1], REGEX_EXTRACT_SINGLE_TERM);
 
-    const words = {} as Record<string, WeightsPerTermPerDocument>;
+    const words: Record<string, TermWeightsPerDocument> = {};
     for (const word of listOfWords) {
       words[word[1]] = {
         Correlation: Number(word[2]),
         'TF-IDF': Number(word[3]),
       };
     }
 
-    const weights = parseWeights(term[2]);
+    const weights = parseWeights<TermWeights>(term[2]);
     terms[Object.keys(words).join(', ')] = {
       terms: words,
-      Weights: (weights as unknown) as WeightsPerTermBreakdown,
+      Weights: weights,
```

Comment:
Removing the cast causes typescript to detect that `weights` can be null, since the `parseWeights` function on line 256 can return this value. With that said, if `term[2]` is always defined, we could avoid this breaking change, since we will always be able to parse an object from the string.